eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java

```diff
@@ -375,6 +380,11 @@ public static Builder newBuilder() {
             return new Builder();
         }
 
+        public Builder withVipAddressResolver(VipAddressResolver vipAddressResolver) {
```

Comment:
Looks like order is important when calling withVipAddressResolver and setVIPAddress.  A better approach would be to cache the value passed to setVIPAddress and call the resolver in build().  You can also set vipAddressResolver = LazyHolder.DEFAULT_VIP_ADDRESS_RESOLVER in the constructor if it's not set right before resolving the address.